### DISCUSSION TITLE
Didn't find it in the docs: approx. level of SQL compatibility?

### DISCUSSION THREAD

**Discussion by metaleap** [asdine/genji#519] [General]
I mean, between Postgres and MariaDB and Oracle and SQL Server there can be wild SQL syntax / expression variations for various constructs. I'm curious what this project is shooting for basically.

In other words: "will an ORM supporting PostgreSQL likely overwhelmingly work well, excepting _(list of non-supported things here... eg. on-update triggers?)_ — or does it also need to have an SQL-generator for eg. SQLite too?" What's your "SQL basis"?

Let's assume we're talking not just CRUD but also `create table`, `create index` (if even applicable?)

Thanks already, really eyeing a giving this a spin instead of CGO SQLite!

**Comment by asdine** [comment]
Starting v0.16, the SQL API will evolve in the following directions:
- Follow as much as possible the SQL-92 standard
- Implement certain PostgreSQL niceties

We are using both Postgres and SQLite as reference implementations but will always have a preference towards Postgres when the API differs.

However, the goal is not to have 100% feature parity with Postgres (I mean, it's Postgres), or even 100% the same API on features that we manage to implement.

> will an ORM supporting PostgreSQL likely overwhelmingly work well, excepting (list of non-supported things here... eg. on-update triggers?)

Short answer: yes, mostly.
